lib/internal/crypto/hash.js

```diff
@@ -140,7 +145,53 @@ Hmac.prototype.digest = function digest(outputEncoding) {
 Hmac.prototype._flush = Hash.prototype._flush;
 Hmac.prototype._transform = Hash.prototype._transform;
 
+function Mac(mac, key, options) {
+  if (!(this instanceof Mac))
+    return new Mac(mac, key, options);
+
+  validateString(mac, 'mac');
+  key = prepareSecretKey(key);
+
+  let nid = EVP_PKEY_HMAC;
+  if (mac === 'poly1305') nid = EVP_PKEY_POLY1305;
+  else if (mac === 'siphash') nid = EVP_PKEY_SIPHASH;
+  else if (mac.startsWith('cmac:')) {
+    nid = EVP_PKEY_CMAC;
+    mac = mac.slice(5);
+  }
+
+  this[kHandle] = new _Mac(nid, toBuf(key), mac);
+  this[kState] = {
+    [kFinalized]: false
+  };
+  LazyTransform.call(this, options);
+}
+
+ObjectSetPrototypeOf(Mac.prototype, LazyTransform.prototype);
+ObjectSetPrototypeOf(Mac, LazyTransform);
+
+Mac.prototype.update = Hash.prototype.update;
+
+Mac.prototype.digest = function digest(outputEncoding) {
+  const state = this[kState];
+  outputEncoding = outputEncoding || getDefaultEncoding();
+
+  if (state[kFinalized]) {
+    const buf = Buffer.from('');
+    return outputEncoding === 'buffer' ? buf : buf.toString(outputEncoding);
+  }
+
+  // Explicit conversion for backward compatibility.
```

Comment:
Do we need this?

Reply:
I think so. For better or worse, code like this currently works:
```js
const h = crypto.createHmac('sha1', 'secret');
const e = { [Symbol.toPrimitive]: () => 'hex' };
h.digest(e);  // '25af6174a0fcecc4d346680a72b7ce644b9a88e8'
```

Reply:
On the other hand, support for that pattern is wildly inconsistent. It doesn't work with `h.update()`, for example. I'll remove it.